src/main/dsl/Regex.kt

```diff
@@ -1,28 +1,47 @@
 package dsl
 
-class Regex : Node() {
-    val expressions = arrayListOf<Node>()
+open class Regex {
+    internal val children: MutableList<Regex> = mutableListOf()
     
-    override fun render(builder: StringBuilder) {
-        // TODO
-        for (e in expressions) {
-            e.render(builder)
-        }
+    override fun toString(): String {
+        return children.joinToString(separator = "")
     }
     
-    fun optional(init: () -> Group): Optional {
-        val node = Optional(init())
-        expressions.add(node)
-        return node
+    fun toRegex(): kotlin.text.Regex {
+        return Regex(this.toString())
     }
     
-    fun literal(init: () -> String): Literal {
-        return Literal(init())
+    fun literal(init: () -> String): Regex {
+        val literal = Literal(init())
+        this.children.add(literal)
+        return this
+    }
+    
+    fun oneOf(init: OneOf.() -> Regex): Regex {
+        val oneOf = OneOf().init()
+        this.children += oneOf
+        return this
+    }
+    
+    fun captureGroup(init: CaptureGroup.() -> Regex): Regex {
+        val captureGroup = CaptureGroup().init()
+        this.children.add(captureGroup)
+        return this
+    }
+    
+    fun regex(init: Regex.() -> Regex): Regex {
```

Comment:
I don't see why we would want to define a regex inside a regex

Reply:
I believe I did this after discovering that you can nest a `regex` block within another `regex` block. I tried to prevent it, but the issue is that the user can still invoke the top-level regex() method to create a regex within a regex. I mean the ideal fix is to have `Regex` completely separate from `Group`, but I think this would lead to a lot of duplication. This along with the above issue with nesting/typechecking should definitely be new issues that are addressed.